src/main/java/org/cryptomator/ui/health/CheckExecutor.java

```diff
@@ -69,24 +69,14 @@ protected Void call() throws Exception {
 				 var cryptor = CryptorProvider.forScheme(vaultConfig.getCipherCombo()).provide(masterkeyClone, csprng)) {
 				c.getHealthCheck().check(vaultPath, vaultConfig, masterkeyClone, cryptor, diagnosis -> {
 					c.getResults().add(Result.create(diagnosis));
-					compareAndSetSeverity(diagnosis.getSeverity());
+					if (highestResultSeverity.compareTo(diagnosis.getSeverity()) < 0) {
+						highestResultSeverity = diagnosis.getSeverity();
+					}
```

Comment:
[Guava to the rescue](https://github.com/google/guava/wiki/CollectionUtilitiesExplained#comparators):

```suggestion
					highestResultSeverity = Comparators.max(highestResultSeverity, diagnosis.getSeverity());
```

Reply:
Fixed in 4a8dbfb